packages/react/src/recipes/seo/Article/Article.module.css

```diff
@@ -28,13 +20,144 @@
   padding: 0;
 }
 
-.heroImageArea {
-  overflow: hidden;
+.subdomainNavBarContainer {
+  width: 100%;
+  position: absolute;
+  z-index: 92;
+  background: var(--brand-color-canvas-default);
+}
+
+.hero {
+  /* TODO Is this the right color, and the right way to override? */
+  --brand-Link-color-default: #ffffff;
+  --brand-color-text-default: #ffffff;
+
+  --parallax-start-spacing: 74px;
+
+  background: black;
+  position: relative;
+  height: var(--hero-height);
+}
+
+.parallax {
+  position: absolute;
+  top: 0;
+  width: 100%;
+  height: var(--hero-height);
+}
+
+.background,
+.foreground {
+  width: 100%;
+  height: 100%;
+
+  animation-fill-mode: forwards;
+  animation-timing-function: linear;
+  animation-timeline: scroll();
+  animation-range: 0vh var(--hero-height);
+}
+
+.background {
+  position: absolute;
+}
+
+.foreground {
+  position: relative;
 }
 
 .heroImage {
-  display: block;
+  min-width: 100%;
+  min-height: 100%;
+}
+
+.heroImageOverlay {
+  position: absolute;
+  top: 0;
+  width: 100%;
+  height: 100%;
+  background: linear-gradient(0deg, rgba(0, 0, 0, 0.8) 0%, rgba(0, 0, 0, 0) 50%, rgba(0, 0, 0, 0.6) 100%);
```

Comment:
The former, but I'm not sure we'd need separate values for light/dark colour modes as the hero styling stays the same across both light and dark; white text, over a semi-transparent black overlay, over an image.

Reply:
> as the hero styling stays the same across both light and dark; white text, over a semi-transparent black overlay, over an image.

👍 thanks Josh.

How are we planning to enforce usage of dark background images through the CMS? Is that something WebExp are onboard with and enforcing further down the pipeline? I think we need to account for light/dark variations of the hero at the systems level, like we do in F2 templates to account for future use-cases also. I can almost guarantee we'll see light backgrounds being used by content teams eventually, and content is impossible to guardrail unless we do it early.

Having only a dark hero also feels a little limiting and inconsistent with how other templates behave. Is that part of design spec? (sorry if I missed this).

If we're set on always having a dark mode, this is fine. Could we just leave a quick comment above that line to remind readers that this is 'based on assumption that hero will only be available in dark color mode' (or similar).

cc. @nsolerieu and @simmonsjenna - in case either of you know too 🙏

Reply:
V good question. My assumption was that we'd always have that white text and dark overlay, regardless of the lightness of the image, but this could be a bad assumption on my part.

At the moment, a light image will look like this (left). The contrast is there, but the dark gradient overlay doesn't look quite right imo, maybe it needs some tweaking. Switching the colours would give us something like this (right) just for reference, which doesn't feel 100% right either, so input on this would be appreciated 🙂


| Current | Colours switched |
|--------|--------|
| ![image](https://github.com/primer/brand/assets/6840861/f4e2d769-373f-4f3d-a948-f2435cd0cb5a) | ![image](https://github.com/primer/brand/assets/6840861/0058e069-242a-4807-b401-aa1d56898cda)
 |

NB: If we're going to use different text/gradient styles on lighter images vs darker images then we'd also need to expose a control in Storybook to allow the user to say whether it is a light or a dark image. We can then look at the value of that control to decide what colours to use.

Reply:
A note on contrast actually. In the first image the contrast of the white text compared to the lightest part of the background is bang on 3:1 which, for large text, [is the minimum value for MAS compliance](https://liquid.microsoft.com/Web/Object/Read/ms.accessibility/Requirements/01.04.03#Zhome). We'd need to be careful not to reduce this further if we tweak the gradient at all

Reply:
@joshfarrant thank you for your suggestions, those are good callouts

I gave these a thought from a design perspective:
- Can we instead suggest to use the phase 1 `boxed` variant for light theme covers? The wallpaper variant is optimized for dark mode visuals
- Can we "force" dark mode in any way - this template can only be used in one color mode?
- The color switched version you suggested blends the hero with the content drawer, could we add a subtle box shadow to the content to create some separation?
<img width="1571" alt="Screenshot 2024-06-12 at 9 49 29 AM" src="https://github.com/primer/brand/assets/10632534/f2745b11-319b-4a4e-a383-524a54470341">

Reply:
Thanks for the input @nsolerieu. If you're happy, I think option 3 would be the simplest approach to implement and maintain, whilst still offering flexibility.

Option 1 would require maintaining two separate templates going forward, so I'd be hesitant to go down that road.

Just to check my own understanding here — do articles have a single cover image (which could be light or dark), or would one article have two cover images; one for light mode and one for dark mode?

Reply:
b14e45c277b4a08c31dc4f2861a3e3896093ea45